LanguageFeatures/Inline-classes/static_analysis_member_invocation_A04_t01.dart

```diff
@@ -0,0 +1,39 @@
+// Copyright (c) 2023, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion If Dm is a method with function type F, and args is omitted, the
+/// invocation has static type [T1/X1 .. Ts/Xs]F.
+///
+/// @description Checks static type of a method tear-off with omitted type
+/// arguments of an `inline` class
+/// @author sgrekhov22@gmail.com
+
+// SharedOptions=--enable-experiment=inline-class
+
+import "../../Utils/static_type_helper.dart";
+
+inline class IC1 {
+  final int id;
+  IC1(this.id);
+
+  int foo<T1, T2 extends num>() => 42;
+}
+
+inline class IC2<T extends num> {
+  final T id;
+  IC2(this.id);
+
+  T foo<T1, T2 extends T>() => id;
+}
+
+main() {
+  IC1 ic1 = IC1(42);
+  ic1.foo.expectStaticType<Exactly<int Function<T1, T2 extends num>()>>();
+
+  IC2 ic2 = IC2<double>(3.14);
```

Comment:
This one is interesting!

We're creating a ghost of type `IC2<double>` (there is no representation of `double` anywhere, because the ghost is an object that doesn't exist, so we have no place to store it), then we're using covariance of the type parameters to assign the ghost (actually: the representation object) to the variable `ic2`, and the type of `ic2` is `IC2<num>`.

Hence, `double` is completely ignored, and all member invocations will be told that `T == num`.

I agree that this should work exactly as shown in line 35. Cool!

Reply:
But perhaps we should use `IC2<num>` explicitly as the type of `ic2`, following the usual rule that the test should test a single thing in a very focused manner, rather than testing i2b as well as the handling of `T` in instance member invocations.

Reply:
Changed to `IC2<num>`